pkg/controller/account/byoc.go

```diff
@@ -121,16 +124,20 @@ func (r *ReconcileAccount) initializeNewCCSAccount(reqLogger logr.Logger, accoun
 			awsv1alpha1.InvalidAccountClaim,
 			awsv1alpha1.ClaimStatusError,
 		)
+		if err != nil {
```

Comment:
We already know that the error is being raised in this block, some logging in the `Validate()` func would help us understand what about the AccountClaim is invalid.

Reply:
The Validate function is part of the API definition, so I don't want to introduce the logger into that package.  Each `validateErr` that should come back should be distinct enough.

We could remove the log message here now that we're explicitly returning the `validateErr` below instead of returning `err`.  Thoughts?